pinot-controller/src/main/java/org/apache/pinot/controller/helix/SegmentStatusChecker.java

```diff
@@ -69,58 +65,50 @@ public SegmentStatusChecker(PinotHelixResourceManager pinotHelixResourceManager,
   }
 
   @Override
-  public void initTask() {
+  protected void setUpTask() {
     LOGGER.info("Initializing table metrics for all the tables.");
     setStatusToDefault();
   }
 
   @Override
-  protected void preprocess() {
-    _realTimeTableCount = 0;
-    _offlineTableCount = 0;
-    _disabledTableCount = 0;
-
+  protected Context preprocess() {
+    Context context = new Context();
     // check if we need to log disabled tables log messages
     long now = System.currentTimeMillis();
     if (now - _lastDisabledTableLogTimestamp >= DISABLED_TABLE_LOG_INTERVAL_MS) {
-      _logDisabledTables = true;
+      context._logDisabledTables = true;
       _lastDisabledTableLogTimestamp = now;
-    } else {
-      _logDisabledTables = false;
     }
+    return context;
   }
 
   @Override
-  protected void processTable(String tableNameWithType) {
-    updateSegmentMetrics(tableNameWithType);
-  }
-
-  @Override
-  protected void exceptionHandler(String tableNameWithType, Exception e) {
```

Comment:
Why remove exceptionHandler method? It was consciously added so that we avoid such try catch in each ControllerPeriodicTask.

Reply:
After consideration, I still think this exception handling logic should be wrapped inside the processTable(). It might need some variables from the processTable(), which are not accessible with this callback handler.